agent/ebs/watcher.go

```diff
@@ -106,12 +124,34 @@ func (w *EBSWatcher) HandleEBSResourceAttachment(ebs *apiebs.ResourceAttachment)
 	volumeId := ebs.GetAttachmentProperties(apiebs.VolumeIdKey)
 	ebsAttachment, ok := w.agentState.GetEBSByVolumeId(volumeId)
 	if ok {
-		log.Infof("EBS Volume attachment already exists. Skip handling EBS attachment %v.", ebs.EBSToString())
+		log.Debugf("EBS Volume attachment already exists. Skip handling EBS attachment %v.", ebs.EBSToString())
 		return ebsAttachment.StartTimer(func() {
 			w.handleEBSAckTimeout(volumeId)
 		})
 	}
 
+	// start EBS CSI Driver Managed Daemon
+	if runningCsiTask := w.taskEngine.GetDaemonTask(md.EbsCsiDriver); runningCsiTask != nil {
+		log.Debugf("engine ebs CSI driver is running with taskID: %v", runningCsiTask.GetID())
+	} else {
+		if ebsCsiDaemonManager, ok := w.taskEngine.GetDaemonManagers()[md.EbsCsiDriver]; ok {
+			if csiTask, err := ebsCsiDaemonManager.CreateDaemonTask(); err != nil {
+				// fail attachment and return
+				log.Errorf("Unable to start ebsCsiDaemon for task %s in the engine: error: %s", csiTask.GetID(), err)
+				csiTask.SetKnownStatus(apitaskstatus.TaskStopped)
+				csiTask.SetDesiredStatus(apitaskstatus.TaskStopped)
+				w.taskEngine.EmitTaskEvent(csiTask, err.Error())
```

Comment:
If `ebsCsiDaemonManager.CreateDaemonTask()` fails with an error then `csiTask` shouldn't be valid and shouldn't be used IMO. It hasn't been added to the task engine so we don't need to update its status anyway, I think? 

For `EmitTaskEvent`, what information is needed? Maybe we can get it by making another method call to `ebsCsiDaemonManager`?

Reply:
(removed)